### PR TITLE
Connection factory goes to cache options

### PR DESCRIPTION
**Pull request by RealFatCat**
Another view for https://github.com/jazzband/django-redis/issues/619

### CODE REVIEW

File: tests/test_connection_string.py [hunk 1]
```diff
@@ -17,3 +17,24 @@ def test_connection_strings(connection_string: str):
...
+def test_connection_strings_with_conn_factory_in_opts(connection_string: str):
+    sentinel_conn_factory = "django_redis.pool.SentinelConnectionFactory"
+    cf = pool.get_connection_factory(
+        path=None,
+        options={
+            "CONNECTION_FACTORY": sentinel_conn_factory,
+            "SENTINELS": [("127.0.0.1", "26739")],
+        },
+    )
+    res = cf.make_connection_params(connection_string)
```
Comment (WisdomPill) [inline tests/test_connection_string.py:39]: why check connection params?

why not check if CF is an instance of `SentinelConnectionFactory`? I would even use multiple connection factories from `parametrize` and check that the same instance is created

**Comment by RealFatCat** [comment]
Added some tests for `get_connection_factory`.

**Comment by akostrikov** [comment]
Wow, nice feature! I really want to see that in production

**Review by WisdomPill — APPROVED** [review]
LGTM!

**Comment by WisdomPill** [comment]
last but not least is to add a changeling file so that when the release is made the `CHANGELOG` is automatic

**Comment by RealFatCat** [comment]
> last but not least is to add a changelog file so that when the release is made the `CHANGELOG` is automatic

done

**Comment by akostrikov** [comment]
@WisdomPill Hello, Anas! Any info on when the next release is going to be?

**Comment by WisdomPill** [comment]
this weekend

**Pull request merged**